.github/workflows/performance.yml

```diff
@@ -0,0 +1,115 @@
+name: Performance tests
+
+on:
+  workflow_call:
+    inputs:
+      go-version:
+        required: true
+        type: string
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  performance-test-suite-detect-runners:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.detect-runners.outputs.matrix }}
+    env:
+      PERF_TEST_RUNS_ON: ${{ secrets.PERF_TEST_RUNS_ON }}
+      PERF_TEST_RUNS_ON_DEFAULT: |
+        {
+          "targets": [
+            {
+              "name": "github-ubuntu-latest",
+              "runs-on": "ubuntu-latest"
+            }
+          ]
+        }
+    steps:
+      - id: detect-runners
+        run: |
+          RES="$(echo "${PERF_TEST_RUNS_ON:-${PERF_TEST_RUNS_ON_DEFAULT}}" | jq -c '.targets')"
+          echo "Detected targets:"
+          echo "$RES" | jq .
+          echo "::set-output name=matrix::${RES}"
+
+  performance-test-suite:
+    name: Performance Test Suite (${{ matrix.target.name }})
+    needs:
+      - performance-test-suite-detect-runners
+    strategy:
+      matrix:
+        target: ${{ fromJson(needs.performance-test-suite-detect-runners.outputs.matrix) }}
+    runs-on: ${{ matrix.target.runs-on }}
+    env:
+      ARG_DURATION: "${{ startsWith(github.ref, 'refs/tags/v') && '-d 10m' || '' }}"
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ inputs.go-version }}
+      - uses: actions/checkout@v3
+      - run: go build -o perf-test-suite ./test/performance-test-suite/cmd/perf-test/
+      - name: Run performance tests
+        id: performance
+        run: |
+          ./perf-test-suite $ARG_DURATION > perf-test-results.json 2>console_output.txt
+          cat console_output.txt
+      - name: Upload test results
+        uses: actions/upload-artifact@v3
+        with:
+          name: Performance Test Results (${{ matrix.target.name }})
+          path: perf-test-results.json
+          retention-days: 30
+      - name: Create the Mattermost message
+        if: github.event.schedule == '0 0 * * *'
+        run: >
+          echo "{\"text\":\"### Results for scheduled daily run on branch ${{ github.base_ref }}\n
+          **Result**: ${{ steps.performance.outcome }}\n
+          #### Benchmark Write TX/s \n
+          $(cat console_output.txt | grep "Results:" | sed -n 1p) \n
```

Comment:
The `perf-test-results.json` file doesn't contain the same value sent to `stdout`, unless I've overlooked something.

Reply:
Hmm, I can see that the json file is broken due to some output from immudb, but if that's cleared then:

```sh
cat perf-test-results.json | jq .benchmarks[].results
```
I think though that the output is more readable. I would only insist on automatically dump all resutlts, e.g.:

```sh
$ cat output.txt | grep -B 1 'Results:'
2022/10/13 16:44:17 Benchmark Write TX/s finished
2022/10/13 16:44:17 Results: TX: 10560, KV: 10560, TX/s: 1055.92, KV/S: 1055.92, CPUTime: 5.63, VMM: 2.14G, RSS: 282.53M, Writes (bytes/calls): 8.03M/1791, Reads (bytes/calls): 4.10k/11328
--
2022/10/13 16:44:28 Benchmark Write KV/s finished
2022/10/13 16:44:28 Results: TX: 2040, KV: 2040000, TX/s: 203.98, KV/S: 203982.94, CPUTime: 34.31, VMM: 3.70G, RSS: 1.44G, Writes (bytes/calls): 474.83M/10177, Reads (bytes/calls): 8.19k/6675
```

Reply:
I don't see the difference on what you're suggesting. As it is, the code I wrote will output something like:

```
### Results for scheduled daily run on branch <branch>
Result:  <step outcome>
#### Benchmark Write TX/s
2022/09/30 11:31:51 Results: TX: 19320, KV: 19320, TX/s: 1931.94, KV/S: 1931.94, CPUTime: 4.03, VMM: 2.21G, RSS: 382.16M, Writes (bytes/calls): 14.69M/3192, Reads (bytes/calls): 8.19k/20612
#### Benchmark Write KV/s
2022/09/30 11:32:02 Results: TX: 1397, KV: 1397000, TX/s: 139.69, KV/S: 139689.08, CPUTime: 23.42, VMM: 3.55G, RSS: 1.40G, Writes (bytes/calls): 338.56M/9892, Reads (bytes/calls): 12.29k/5443
```

Reply:
The difference is in maintainability only - if we add new test case to perf test suite, then it will/may cause following issues:
- Results of the new test will not show up on slack notification
- Results can be reordered so wrong title will be applied to results

With `grep -B1` we will get this sorted out automatically.

Reply:
I've updated the pull request with the changes we discussed in a call and making use of the changes you made after it.